sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/GroupByKey.java

```diff
@@ -244,6 +264,20 @@ public PCollection<KV<K, Iterable<V>>> expand(PCollection<KV<K, V>> input) {
       throw new IllegalStateException("the keyCoder of a GroupByKey must be deterministic", e);
     }
 
+    PipelineOptions options = input.getPipeline().getOptions();
+    String gbekOveride = options.getGBEK();
+    if (!this.insideGBEK && gbekOveride != null && !gbekOveride.trim().isEmpty()) {
```

Comment:
same, consider `Strings.isNullOrEmpty`

Reply:
If I do this here (and in DataflowGroupByKey), it doesn't compile because we don't have the explicit null check:

```
/Users/dannymccormick/beam/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/GroupByKey.java:272: error: [argument] incompatible argument for parameter secretOption of Secret.parseSecretOption.
      Secret hmacSecret = Secret.parseSecretOption(gbekOveride);
```

It seems cleaner to me to leave as is vs working around it.